apps/server/src/task/task.controller.ts

```diff
@@ -1,4 +1,123 @@
-import { Controller } from '@nestjs/common';
+import {
+  BadRequestException,
+  Body,
+  Controller,
+  Get,
+  InternalServerErrorException,
+  Logger,
+  Post,
+  Query,
+  DefaultValuePipe,
+  ParseIntPipe,
+} from '@nestjs/common';
+import { TaskService } from './task.service';
+import { TaskDto, EditTaskReqDto, UpdateTaskReqDto } from './dto/task.dto';
+import { UserJwtPayload } from '../user/dto/jwt-user.dto';
+import { User } from '../decorators/User';
 
 @Controller('task')
-export class TaskController {}
+export class TaskController {
+  constructor(private readonly taskService: TaskService) {}
+  private readonly logger = new Logger(TaskController.name);
+
+  @Post('create')
+  async createTask(@Body() taskDto: TaskDto) {
+    // TODO add validation
+    //TODO make sure assigned users belong to the apartment
+    try {
+      const task = await this.taskService.createTask(taskDto);
+      return { success: true, task };
+    } catch (error) {
+      this.logger.error('Error in createTask:', error.stack);
+      throw new InternalServerErrorException('Failed to create task', {
+        description: 'לא הצלחנו ליצור את המשימה. אנא נסו בשנית',
+      });
+    }
+  }
+
+  /* Only task owner can edit the task.
+  Task participants can only update if they have completed the task */
+  @Post('update')
+  async updateTaskStatus(@User() user: UserJwtPayload, @Body() dto: UpdateTaskReqDto) {
+    const userId = user.userId;
+    const task = await this.taskService.getTaskById(dto.taskId);
+    let completedTask = true;
+
+    if (!task) {
+      throw new BadRequestException('Task was not found');
+    }
+    task.completions.forEach(participant => {
+      if (!participant.isCompleted) {
+        completedTask = false;
+      }
+    })
```

Comment:
```suggestion
    const isTaskComplete = task.completions.every(participant => 
      participant.isCompleted
    )
```
אני חושבת שעדיף להשתמש בפונקציות של JS אז רק תבדוק ששמרתי על לוגיקה נכונה. אבל אפשר גם להתעלם מההצעה שלי זה בסדר